### PR TITLE
fix: #293 - Cleanup Warnings (Partial)

### PR DESCRIPTION
**Pull request by Christian-Oleson**
# Pull Request

## What does this PR do?
This is my (partial) attempt to fix the issue where warnings were occurring (as well as errors in the console). I got to a bit of a standstill, as I haven't written react in a very long time. I got stuck cleaning up the very last warning, and the console error I was not familiar with enough to resolve. This at least gets the issue closer to being done. Though, I'm completely fine if this acts as a starting point for someone else to complete (e.g. clone my branch, resolve the two issues I struggled with, and then get it merged.)

 There were many != instead of !== operators, which were fixed.
 There were several unnecessary assignments as well as imports
 
 My motivation to do this was just to step in an learn some react. I'm a back-end dev, so I wanted to see how far I could get (I last wrote React 2 years ago)
 No dependencies are required to be changed

The following links the related issue to the PR (https://github.com/Auto-DL/Auto-DL/issues/293)

## What part does this affect?
- [X] FrontEnd.
- [ ] BackEnd.
- [ ] Documentation.
- [ ] Other. (Please specify below)

- FrontendApp v1 React only

## Before submitting
- [X] Was this discussed/approved via a GitHub issue or slack?
- [X] Did you read the [contributor guideline](https://github.com/Auto-DL/Auto-DL/blob/v1-beta/CONTRIBUTING.md)?
- [X] Did you ensure that there aren't any other open [Pull Requests](https://www.github.com/Auto-DL/Generator/pulls) for the same update/change?
- [X] Did you make sure the title is self-explanatory and the description concisely explains the PR?
- [X] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [X] Did you make sure the code is clean and docstrings have been added or updated as required?
- [X] Did you make sure the code is linted/formatted locally prior to submission? (using `black` and/or `prettier`)
- [N/a] Did you make sure to update the documentation with your changes? (if necessary)

## PR review
Anyone in the community is free to review the PR once the tests have passed.

### CODE REVIEW

**Comment by RusherRG** [comment]
@Christian-Oleson Thanks a lot for solving the errors and warnings. It would be great if you could mention the warnings that are still present

**Comment by Christian-Oleson** [comment]
> @Christian-Oleson Thanks a lot for solving the errors and warnings. It would be great if you could mention the warnings that are still present

There are a few issues still present:
`Line 120:9:  The 'getProjectId' function makes the dependencies of useEffect Hook (at line 2210) change on every render. To fix this, wrap the definition of 'getProjectId' in its own useCallback() Hook  react-hooks/exhaustive-deps`
and
`Warning: findDOMNode is deprecated in StrictMode. findDOMNode was passed an instance of Transition which is inside StrictMode. Instead, add a ref directly to the element you want to reference. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-find-node`

all of the other warnings are gone. I cannot find a way to assign to myself, though normally I can just click the "assign" button. It seems it is disabled for me.

**Comment by RusherRG** [comment]
> There are a few issues still present: `Line 120:9: The 'getProjectId' function makes the dependencies of useEffect Hook (at line 2210) change on every render. To fix this, wrap the definition of 'getProjectId' in its own useCallback() Hook react-hooks/exhaustive-deps` and `Warning: findDOMNode is deprecated in StrictMode. findDOMNode was passed an instance of Transition which is inside StrictMode. Instead, add a ref directly to the element you want to reference. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-find-node`

Thanks we'll have a look at it. 

> I cannot find a way to assign to myself, though normally I can just click the "assign" button. It seems it is disabled for me.

That's weird. Just try adding a comment to the issue. I'll try if I can assign you.

File: FrontEndApp/v1-react/src/Components/MainApp/Step2.js [hunk 1]
```diff
@@ -78,7 +78,7 @@ function Step2() {
   const [components, setcomponents] = useState([]);
   const [selected_layer_type, setselected_layer_type] = useState("");
   const [selected_layer, setselected_layer] = useState(-1);
-  const [selected_layer_name, setselected_layer_name] = useState("");
+  const [setselected_layer_name] = useState("");
```
Comment (McTechie) [inline FrontEndApp/v1-react/src/Components/MainApp/Step2.js:81]:
```suggestion
  const [, setselected_layer_name] = useState("");
```

File: FrontEndApp/v1-react/src/Components/MainApp/Step2.js [hunk 2]
```diff
@@ -98,7 +98,7 @@ function Step2() {
   const [openModal, setOpenModal] = useState(false);
   const [openGitHubDetails, setOpenGitHubDetails] = useState(false);
 
-  const [generated_file_path, setgenerated_file_path] = useState("");
+  const [setgenerated_file_path] = useState("");
```
Comment (McTechie) [inline FrontEndApp/v1-react/src/Components/MainApp/Step2.js:101]:
```suggestion
  const [, setgenerated_file_path] = useState("");
```

File: FrontEndApp/v1-react/src/Components/MainApp/Step2.js [hunk 3]
```diff
@@ -2138,13 +2139,13 @@ function Step2() {
     };
   }
 
-  const [all_optimizer, setall_optimizer] = useState(temp_optimizer);
-  const [all_loss, setall_loss] = useState(temp_loss);
+  const [all_optimizer] = useState(temp_optimizer);
```
Comment (McTechie) [inline FrontEndApp/v1-react/src/Components/MainApp/Step2.js:2142]:
```suggestion
  const [all_optimizer, ] = useState(temp_optimizer);
```

File: FrontEndApp/v1-react/src/Components/MainApp/Step2.js [hunk 4]
```diff
@@ -2138,13 +2139,13 @@ function Step2() {
...
+  const [all_loss] = useState(temp_loss);
```
Comment (McTechie) [inline FrontEndApp/v1-react/src/Components/MainApp/Step2.js:2143]:
```suggestion
  const [all_loss, ] = useState(temp_loss);
```

File: FrontEndApp/v1-react/src/Components/MainApp/Step2.js [hunk 5]
```diff
@@ -2138,13 +2139,13 @@ function Step2() {
...
   const [all_prepro, setall_prepro] = useState({});
-  const [render_prepro_meta, setrender_prepro_meta] = useState(temp_pre_meta);
-  const [render_prepro, setrender_prepro] = useState(temp_pre);
+  const [render_prepro_meta] = useState(temp_pre_meta);
```
Comment (McTechie) [inline FrontEndApp/v1-react/src/Components/MainApp/Step2.js:2145]:
```suggestion
  const [render_prepro_meta, ] = useState(temp_pre_meta);
```

File: FrontEndApp/v1-react/src/Components/MainApp/Step2.js [hunk 6]
```diff
@@ -2138,13 +2139,13 @@ function Step2() {
...
+  const [render_prepro] = useState(temp_pre);
```
Comment (McTechie) [inline FrontEndApp/v1-react/src/Components/MainApp/Step2.js:2146]:
```suggestion
  const [render_prepro, ] = useState(temp_pre);
```

File: FrontEndApp/v1-react/src/Components/MainApp/Step2.js [hunk 7]
```diff
@@ -2138,13 +2139,13 @@ function Step2() {
...
   const [show_pre, setshow_pre] = useState(false);
-  const [jsondata, setjsondata] = useState(temp_json);
+  const [jsondata] = useState(temp_json);
```
Comment (McTechie) [inline FrontEndApp/v1-react/src/Components/MainApp/Step2.js:2148]:
```suggestion
  const [jsondata, ] = useState(temp_json);
```

**Review by McTechie — CHANGES_REQUESTED** [review]
While de-structuring from the `useState()` hook it would be better to leave out the unused variables / functions, rather than completely omitting them! xD

**Comment by bamblebam** [comment]
@Christian-Oleson can you comment on the related issue so I can assign it to you?

**Comment by Christian-Oleson** [comment]
> @Christian-Oleson can you comment on the related issue so I can assign it to you?

Done!

**Comment by Christian-Oleson** [comment]
> While de-structuring from the `useState()` hook it would be better to leave out the unused variables / functions, rather than completely omitting them! xD

Thanks for the review! I learned something new after your suggestion. 😄

**Review by McTechie — APPROVED** [review]
Looks good to me! xD

**Pull request merged**